os/drivers/ai-soc/ndp120/src/ndp120_api.c

```diff
@@ -1345,25 +1345,44 @@ int ndp120_extract_audio(struct ndp120_dev_s *dev, struct ap_buffer_s *apb)
 		return SYNTIANT_NDP_ERROR_NONE;
 	}
 
-	/* wait for sample interrupt */
+	/* wait for sample interrupt for a given time */
 
 	int err = pthread_mutex_lock(&dev->ndp_mutex_notification_sample);
 	if (err) {
 		auddbg("NDP sample mutex lock err: %d\n", err);
 		return SYNTIANT_NDP_ERROR_FAIL;
 	}
-	err = pthread_cond_wait(&dev->ndp_cond_notification_sample,
-							&dev->ndp_mutex_notification_sample);
-	if (err) {
-		auddbg("NDP sample wait err: %d\n", err);
-		return SYNTIANT_NDP_ERROR_FAIL;
+
+	struct timespec abstime;
+	clock_gettime(CLOCK_REALTIME, &abstime);
+	/* Set timeout value to three sample sizes (2 should be enough). Generally the time values are in milli seconds */
+	/* 16KHz, 1 channel, 2bytes = 16000 * 1 * 2 = 32000 bytes per second, 32 bytes per millisecond, (sample size is 640 or 768) */
+	uint32_t timeout_in_msec = (3 * dev->sample_size * 1000) / (SYNTIANT_NDP120_AUDIO_SAMPLE_RATE * SYNTIANT_NDP120_AUDIO_SAMPLES_PER_WORD);
+	abstime.tv_nsec += timeout_in_msec * 1000000;
```

Comment:
Do we need to check for roll over here? If tv_nsec rolls over, then we need to increment tv_sec.

Reply:
Hello Kishore, this is a sync PR from internal branch. I will take care of this comment in a different PR.